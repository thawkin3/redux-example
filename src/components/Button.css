button {
	margin: 5px;
}
@media screen and (max-width: 480px) {
	button {
		display: block;
		margin: 10px auto;
	}
}