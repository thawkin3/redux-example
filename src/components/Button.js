import React from 'react';
import './Button.css';

const Button = ({ buttonText, clickHandler }) => (
    <button
        type="button"
        onClick={clickHandler}
    >
        {buttonText}
    </button>
);

Button.displayName = 'Button';

export default Button;
