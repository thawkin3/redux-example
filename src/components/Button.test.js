import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Button } from './Button';

let wrapper, props;

describe('Button', () => {
	beforeEach(() => {
		props = {
			buttonText: 'Click me!',
			clickHandler: () => { /* do nothing */ },
		};
		wrapper = shallow(<Button {...props} />);
	});

	it('renders a single button', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('renders a button element', () => {
		expect(wrapper.find('button').length).toEqual(1);
	});

	it('matchs snapshot', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	})

	it('shows the correct button text', () => {
		expect(wrapper.find('button').prop('children')).toEqual('Click me!');
	});

	it('calls the click handler when clicked', () => {
		const spy = jest.spyOn(props, 'clickHandler');
		wrapper = shallow(<Button {...props} />);
		const mockEvent = {
			target: {
				value: 'Hey!'
			}
		};
		wrapper.find('button').simulate('click', mockEvent);
		expect(spy).toHaveBeenCalledTimes(1);
		spy.mockRestore();
	});

	it('renders without crashing', () => {
	    const div = document.createElement('div');
	    ReactDOM.render(<Button {...props} />, div);
	    ReactDOM.unmountComponentAtNode(div);
	});


});	