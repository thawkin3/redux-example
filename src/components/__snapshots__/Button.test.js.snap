// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matchs snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Click me!
</button>
`;
